src/java/org/apache/cassandra/service/opstate/KeyspaceTableOpStatePersister.java

```diff
@@ -34,127 +30,94 @@
         {
+            logIoException("Failed to read state from file.", persistentFileLocation.toFile().getAbsolutePath(), e);
             if (e instanceof JsonParseException)
             {
                 log.warn("Persistent file corrupted, wiping content.");
-                writeStateToFile(persistentStateFile, ImmutableMap.of());
+                writeStateToFile(ImmutableMap.of());
             }
             return Optional.empty();
         }
     }
 
     public boolean updateStateInPersistentLocation(Map<KeyspaceTableKey, Instant> updatedEntries)
     {
-        File persistentStateFile = getOrMaybeCreateStateFile();
-        if (persistentStateFile == null)
-            return false;
-
-        return writeStateToFile(persistentStateFile, updatedEntries);
-    }
-
-    private File getOrMaybeCreateStateFile()
-    {
-        if (persistentFile != null)
-            return persistentFile;
-
-        File operationStateFile = persistentFileLocation.toFile();
         try
         {
-            operationStateFile.createNewFile();
+            parser.create();
+            return writeStateToFile(updatedEntries);
         }
         catch (IOException e)
         {
-            log.warn("Cannot retrieve or create state file.", operationStateFile.getAbsolutePath(), e);
-            return null;
+            logIoException("Cannot retrieve or create state file.", persistentFileLocation.toFile().getAbsolutePath(), e);
+            return false;
         }
-        this.persistentFile = operationStateFile;
-        return operationStateFile;
     }
 
-    private Map<KeyspaceTableKey, Instant> readStateFromFile(File file) throws IOException
+    private boolean writeStateToFile(Map<KeyspaceTableKey, Instant> updatedEntries)
     {
-        if(file.length() == 0)
-            return new HashMap<>();
         try
         {
-            return convertMapTypeToKeyspaceTableKeyInstant(OBJECT_MAPPER.readValue(file, Map.class));
+            parser.write(convertMapTypeToStringLong(updatedEntries));
+            return true;
         }
         catch (IOException e)
         {
-            log.warn("Failed to read state from file.", file.getAbsolutePath(), e);
-            throw e;
+            return false;
         }
     }
 
-    private boolean writeStateToFile(File file, Map<KeyspaceTableKey, Instant> updatedEntries)
+    private void init()
     {
         try
         {
-            atomicWritetoFile(file, OBJECT_MAPPER.writeValueAsString(convertMapTypeToStringLong(updatedEntries)));
-            return true;
+            parser.create();
         }
         catch (IOException e)
         {
-            return false;
+            logIoException("Failed to create state file.", persistentFileLocation.toFile().getAbsolutePath(), e);
         }
     }
 
-    private synchronized void atomicWritetoFile(File file, String content) throws IOException
+    private void logIoException(String message, String path, IOException e)
```

Comment:
`log.warn` is actually shorter than `logIoException`? Any reason not to inline?

Reply:
I intended for it to be an easy way to switch log level.

Reply:
Unless you think we'll soon need to modify log level, let's inline for now. We can always reconsider later